dms/models/ir_attachment.py

```diff
@@ -0,0 +1,85 @@
+import re
+
+from odoo import api, models
+
+
+class IrAttachment(models.Model):
+
+    _inherit = "ir.attachment"
+
+    def _dms_file_name(self, directory_id, file_name):
+        count = 1
+        for file in directory_id.file_ids:
+
+            dms_file_name = re.sub(r" \([0-9]*\)", "", file.name)
+            if dms_file_name == file_name:
+                count += 1
+
+        file, ext = re.split(r"\.", file_name)
+        return file + " ({}).".format(count) + ext if count > 1 else file_name
+
+    def _create_record_dir(self, storage_id, attachment_id):
+        dms_field = self.env["ir.module.module"].search([("name", "=", "dms_field")])
+
+        directory_fields = {
+            "name": attachment_id.res_name.replace("/", "-"),
+            "is_root_directory": True,
+            "parent_id": False,
+            "root_storage_id": storage_id.id,
+        }
+
+        if dms_field and dms_field.state == "installed":
+            directory_fields["res_id"] = attachment_id.res_id
```

Comment:
IMO it's unnecessary because of (in the future) other addon use it, it's not necessary to add new field `res_id` in directory because `record_ref` contain it already.

Reply:
I think the standard is to have `res_model` and `res_id` field. The `record_ref` can be a computed one, composed over the other 2.

Reply:
Ok, but actually don't exists this fields.
Do you think that it's necessary to remove `record_ref` and add: `res_model` and `res_id` fields?
It's necessary to add this fields only in directory or in file too? (now group all attachment in only one directory, it's enought?)

Reply:
I think it's better to go with `res_model` and `res_id` fields, and the other one computed (stored). If `res_model` can be computed stored as well from the `dms.directory`, perfect. I understand that `dms.directory` can contain `res_model` and optionally `res_id`, isn't it?